open-weather-connector-demo/src/com/axonivy/connector/openweather/dto/DailyForecast.java

```diff
@@ -0,0 +1,99 @@
+package com.axonivy.connector.openweather.dto;
+
+import java.time.LocalDate;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+import org.openweathermap.api.data2_5.client.WeatherRecord;
+
+import com.axonivy.connector.openweather.util.Constants;
+
+public class DailyForecast {
+	private final LocalDate date;
+	private final List<WeatherRecord> dailyRecords;
+	private final int temperature;
+	private final int minTemperature;
+	private final int maxTemperature;
+	private final int humidity;
+	private final float windSpeed;
+	private final String weatherIcon;
+	private final String weatherDescription;
+
+	public DailyForecast(LocalDate date, List<WeatherRecord> dailyRecords) {
+		this.date = date;
+		this.dailyRecords = dailyRecords;
+
+		this.temperature = dailyRecords.stream().map(record -> record.getMain().getTemp()).filter(Objects::nonNull)
+				.mapToInt(Float::intValue).max().orElse(Constants.DEFAULT_WEATHER_TEMPERATURE_DEGREE);
+
+		this.maxTemperature = dailyRecords.stream().map(record -> record.getMain().getTempMax())
+				.filter(Objects::nonNull).mapToInt(Float::intValue).max()
+				.orElse(Constants.DEFAULT_WEATHER_TEMPERATURE_DEGREE);
+
+		this.minTemperature = dailyRecords.stream().map(record -> record.getMain().getTempMin())
+				.filter(Objects::nonNull).mapToInt(Float::intValue).min()
+				.orElse(Constants.DEFAULT_WEATHER_TEMPERATURE_DEGREE);
+
+		this.humidity = dailyRecords.stream().map(record -> record.getMain().getHumidity()).max(Integer::compareTo)
+				.orElse(Constants.DEFAULT_WEATHER_HUMIDITY);
+
+		this.windSpeed = dailyRecords.stream().map(record -> record.getWind().getSpeed()).max(Float::compareTo)
+				.orElse(Constants.DEFAULT_WEATHER_WIND_SPEED);
+
+		WeatherRecord maxPriorityRecord = dailyRecords.stream()
+				.max(Comparator.comparingInt(record -> getWeatherPriority(record.getWeather().get(0).getId())))
+				.orElse(null);
+
+		if (maxPriorityRecord != null) {
+			this.weatherIcon = maxPriorityRecord.getWeather().get(0).getIcon();
+			this.weatherDescription = maxPriorityRecord.getWeather().get(0).getDescription();
+		} else {
+			this.weatherIcon = Constants.DEFAULT_WEATHER_CONDITION_ICON;
+			this.weatherDescription = Constants.DEFAULT_WEATHER_CONDITION_DESCRIPTION;
+		}
+	}
+
+	public LocalDate getDate() {
+		return date;
+	}
+
+	public List<WeatherRecord> getDailyRecords() {
+		return dailyRecords;
+	}
+
+	public int getTemperature() {
+		return temperature;
+	}
+
+	public int getMaxTemperature() {
+		return maxTemperature;
+	}
+
+	public int getMinTemperature() {
+		return minTemperature;
+	}
+
+	public int getHumidity() {
+		return humidity;
+	}
+
+	public float getWindSpeed() {
+		return windSpeed;
+	}
+
+	public String getWeatherIcon() {
+		return weatherIcon;
+	}
+
+	public String getWeatherDescription() {
+		return weatherDescription;
+	}
+
+	private static int getWeatherPriority(int weatherId) {
```

Comment:
This function should be part of class rather than instance. It look like helper or util, but I don't want to create a class just contains only function so... I put it here

Reply:
I understand what you mean, but I think this function should be static because it seems more class method than instance method.